main.go

```diff
@@ -20,8 +21,15 @@ import (
 	"github.com/whyrusleeping/stump"
 )
 
+//go:embed  version.json
+var versionFile []byte
+
 func init() {
 	stump.ErrOut = os.Stderr
+	err := config.SetVersionNumber(versionFile)
```

Comment:
This doesn't follow dependency analysis rules and break if someone use `CurrentVersionNumber` at init time.
(I know this isn't happening currently I just don't like surprises)
You should move `CurrentVersionNumber` in this package and do something like `var CurrentVersionNumber = readCurrentVersionNumberFromEmbed()`.

Reply:
@Jorropo would https://github.com/ipfs/ipfs-update/pull/167/commits/8edeb6dd835deaffda8904f7fcb37637b00c5fa9 be ok? AFAIK `config.go` is 7-year-old copy-paste from go-ipfs, never used for anything.

Reply:
perfect